libutils/json-utils.c

```diff
@@ -0,0 +1,263 @@
+ * @brief Filters a string, according to env file format
+ *
+ * This is used to parse the part after the equal sign in an env file
+ * Leading and trailing whitespace should already be removed
+ * Will terminate string after closing double quote, even when there's more
+ *
+ * @param src Copy from pointer, can be same as dst
+ * @param dst Copy to pointer, can be same as src
+ * @return beginning of processed string, either dst or dst + 1
+ */
+static char *filtered_copy(const char *src, char *dst)
+{
+    assert(src);
+    assert(dst);
+    char *beginning = dst;
+    char opening_quote = '\0';
+    while (*src != '\0')
+    {
+        if (*src == '\"' || *src == '\'')
+        {
+            if (opening_quote == *src)
+            {
+                break;
+            }
+            else if (opening_quote == '\0')
+            {
+                opening_quote = *src;
+                ++src;
+                if (*src == opening_quote)
+                {
+                    break;
+                }
+            }
+        }
+        if (*src == '\\')
+        {
+            ++src;
+        }
+        *dst = *src;
+        ++src;
+        ++dst;
+    }
+    *dst = '\0';
+    return beginning;
+}
+
+/**
+ * @brief Splits a line of en env file into key and value string
+ *
+ * See filtered_copy for details on how value in (key-value pair) is parsed
+ * Empty lines are skipped
+ * Lines with first nonspace symbol '#' are skipped (comments)
+ * To preserve whitespace use quotes: WHITESPACE="   "
+ *
+ * @param raw_line input from CfReadLine. Will be edited!
+ * @param key_out   Where to store pointer to key in raw_line
+ * @param value_out Where to store pointer to value in raw_line
+ * @param filename_for_log Optional name for logging purposes
+ */
+ void ParseEnvLine(char *raw_line, char **key_out, char **value_out, const char *filename_for_log)
+ {
+    assert(raw_line);
+    assert(key_out);
+    assert(value_out);
+    char *key = NULL;
+    char *value = NULL;
+    *key_out = NULL;
+    *value_out = NULL;
+
+    char *line = TrimWhitespace(raw_line);
+    if (NULL_OR_EMPTY(line))
+    {
+        return;
+    }
+
+    const char *myname = "ParseEnvLine";
+    int line_length = strlen(line);
+
+    if (line[0] == '#' || line_length == 0)
+    {
+        return;
+    }
+    char *next_equal = strchr(line, '=');
+    if (next_equal == NULL)
+    {
+        Log(LOG_LEVEL_DEBUG, "%s: The line '%s' in ENV file '%s' isn't empty, but was skipped because it's missing an equal sign",
+            myname, line, filename_for_log);
+        return;
+    }
+    int equal_index = (int)(next_equal - line);
+    if (equal_index == 0)
+    {
+        Log(LOG_LEVEL_DEBUG, "%s: The line '%s' in ENV file '%s' was skipped because it's missing a key",
+            myname, line, filename_for_log);
+        return;
+    }
+    *next_equal = '\0';
+    key = TrimWhitespace(line);
+    value = TrimWhitespace(next_equal + 1);
+    value = filtered_copy(value, value);
```

Comment:
What? This?
```
value = filtered_copy(value, value);
```
Copy from value to value, returning the new starting pos in value(dest) :)